smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -4,56 +4,112 @@
-    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer>, java.io.Serializable {
 
+    private static String defaultRegion = Locale.getDefault().getCountry();
     private String telephoneNumber;
 
     /**
-     * @param telephoneNumber Address for the peer.
-     * @throws InvalidTelephoneNumberException If telephoneNumber check is not {@link TelephoneNumberState#TELEPHONE_NUMBER_VALID}.
+     * Set the default country code used when a telephone number without prefix is given.
+     *
+     * @param defaultRegion The default 2 digit ISO country code.
+     * @throws IllegalArgumentException If the given code is not a valid ISO country code.
+     * @author Matteo Carnelos
      */
-    public SMSPeer(String telephoneNumber) throws InvalidTelephoneNumberException {
-        TelephoneNumberState telephoneNumberState = SMSPeer.checkPhoneNumber(telephoneNumber);
-        if(telephoneNumberState != TelephoneNumberState.TELEPHONE_NUMBER_VALID)
-            throw new InvalidTelephoneNumberException(telephoneNumberState);
+    public static void setDefaultRegion(@NonNull String defaultRegion) {
+        if(!Arrays.asList(Locale.getISOCountries()).contains(defaultRegion))
+            throw new IllegalArgumentException("The given string is not a valid ISO country code");
+        SMSPeer.defaultRegion = defaultRegion;
+    }
 
-        this.telephoneNumber = telephoneNumber;
+    /**
+     * Create a new SMSPeer object. If the given number does not explicitly contains the prefix
+     * will be added the default one.
+     * Note: It is not always possible to obtain the correct default country code, especially if the
+     * phone is moving between countries. To set a custom one use the method {@link #setDefaultRegion(String)}.
+     *
+     * @param telephoneNumber Address for the peer.
+     * @throws InvalidTelephoneNumberException If telephoneNumber is not valid.
+     * @throws IllegalArgumentException If the given telephoneNumber is not parsable.
+     * @author Matteo Carnelos
+     */
+    public SMSPeer(@NonNull String telephoneNumber) throws InvalidTelephoneNumberException {
+        // If the given number is an emulator number, skip the validity checks
+        if(adjustForEmulator(telephoneNumber)) return;
+        if(defaultRegion.isEmpty() && !telephoneNumber.startsWith("+"))
+            throw new InvalidTelephoneNumberException("The given number does not explicitly contain " +
+                    "the prefix and the default region is not obtainable. Please use the method " +
+                    "SMSPeer.setDefaultRegion() in order to set a valid one.");
+        Phonenumber.PhoneNumber phoneNumber = parseNumber(telephoneNumber);
+        if(!PhoneNumberUtil.getInstance().isValidNumber(phoneNumber))
+            throw new InvalidTelephoneNumberException("Invalid telephone number: \"" + telephoneNumber + "\". " +
```

Comment:
I suggest putting both the exception messages as static final fields at the top of the class. For this second message, requiring a parameter, you could look into String.format

Reply:
I agree

Reply:
I will definitely do it, thank you